### ISSUE TITLE
Add FAEST

### ISSUE THREAD

**Issue by mkannwischer** [mupq/pqm4#271]

**Comment by mkannwischer** [comment]
Lots of dynamic memory allocations: https://github.com/search?q=repo%3Afaest-sign%2Ffaest-ref%20alloc&type=code

This is going to be some work to make it work on the M4.

**Comment by mkannwischer** [comment]
I revisted FAEST just now to see if we can maybe eliminate the dynamic memory allocations, but it is too heavily used and pointers are returned over multiple levels. I hope that the FAEST team will provide an implementation without the use of dynamic memory allocation at a later point.

**Comment by mkannwischer** [comment]
I revisted FAEST again. Most of the ca. 160 dynamic memory allocations are easily replaced by VLAs, but I found at least one case where more work is required: 
```
typedef struct vec_com_t {
  uint8_t* h;
  uint8_t* k;
  uint8_t* com;
  uint8_t* sd;
} vec_com_t;
```
Here the the `com` and `sd` fields depend on the `numVoleInstances` which can either be the parameter `K0` or `K1`. 
There is a couple of other places where I supect similar problems (e.g., `vec_com_rec_t`, `tree_t`).
Not going to continue with this.
